x/proof/keeper/msg_server_submit_proof.go

```diff
@@ -1,22 +1,56 @@
 package keeper
 
 import (
+	"bytes"
 	"context"
+	"crypto/sha256"
 
+	"github.com/pokt-network/smt"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 
+	"github.com/pokt-network/poktroll/pkg/relayer/protocol"
 	"github.com/pokt-network/poktroll/x/proof/types"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sessionkeeper "github.com/pokt-network/poktroll/x/session/keeper"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
 )
 
+const (
+	// relayMinDifficultyBits is the minimum difficulty that a relay must have
+	// to be reward (i.e. volume) applicable.
+	// TODO_BLOCKER: relayMinDifficultyBits should be a governance-based parameter
+	relayMinDifficultyBits = 0
+
+	// sumSize is the size of the sum of the relay request and response in bytes.
+	// This is needed to extract the relay request and response from the closest
+	// merkle proof.
+	// TODO_TECHDEBT(@h5law): Add a method on the smst to extract the value hash
+	// or export sumSize to be used instead of current local value
+	sumSize = 8
+)
+
+// SMT specification used for the proof verification.
+var spec *smt.TrieSpec
+
+func init() {
+	// Use a spec that does not prehash values in the smst. This returns a nil
+	// value hasher for the proof verification in order to to avoid hashing the
+	// value twice.
+	spec = smt.NoPrehashSpec(sha256.New(), true)
```

Comment:
Nothing actionable right now but this won't be needed anymore after https://github.com/pokt-network/smt/pull/42 merges you can just pass the spec normally.